lock/src/main/java/com/auth0/android/lock/utils/EnterpriseConnectionMatcher.java

```diff
@@ -0,0 +1,131 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.auth0.android.lock.utils;
+
+import android.support.annotation.Nullable;
+
+import com.auth0.Auth0Exception;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Helper class to find out which email domains can be valid for the current Auth0 configuration.
+ */
+public class EnterpriseConnectionMatcher {
+
+    private static final String DOMAIN_KEY = "domain";
+    private static final String DOMAIN_ALIASES_KEY = "domain_aliases";
+
+    private List<Strategy> strategies;
+
+    public EnterpriseConnectionMatcher(List<Strategy> strategies) {
+        this.strategies = new ArrayList<>();
+        if (strategies == null) {
+            return;
+        }
+
+        for (Strategy s : strategies) {
+            if (s.getType() == Strategies.Type.ENTERPRISE) {
+                this.strategies.add(s);
+            }
+        }
+    }
+
+    /**
+     * Tries to find a valid domain with the given input.
+     *
+     * @param email to search the Domain for.
+     * @return a Connection if found, null otherwise.
+     */
+    @Nullable
+    public Connection parse(String email) {
+        String domain = extractDomain(email);
+        if (domain.isEmpty()) {
+            return null;
+        }
+
+        domain = domain.toLowerCase();
+        for (Strategy s : strategies) {
+            for (Connection c : s.getConnections()) {
+                String mainDomain = c.getValueForKey(DOMAIN_KEY);
+                if (mainDomain != null && mainDomain.contains(domain)) {
+                    return c;
+                }
+
+                List<String> aliases = c.getValueForKey(DOMAIN_ALIASES_KEY);
+                if (aliases != null) {
+                    for (String d : aliases) {
+                        if (d.contains(domain)) {
+                            return c;
+                        }
+                    }
+                }
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Extracts the domain part from the email
+     *
+     * @param email to parse
+     * @return the domain String if found, an empty String otherwise
+     */
+
+    public String extractDomain(String email) {
```

Comment:
@hzalaz  To test the domain extraction it needs a public scope. Do you prefer to test just the `parse` method?